gemrb/core/CharAnimations.cpp

```diff
@@ -2801,6 +2801,8 @@ void CharAnimations::PulseRGBModifiers()
 
 static inline void applyMod(const Color& src, Color& dest, const RGBModifier& mod) noexcept
 {
+	dest.a = 0xFF;
```

Comment:
I guess this is just from the default init of Color? It'd be better if we fixed this at the source.
I see `BMPImporter::GetPalette`, PNGImporter and PVRZImporter already handle it properly, just BAMImporter is missing the same signature, since it's not an ImgMgr. But it has code in `BAMImporter::Import` that should take care of it — is the v2 alpha detection at the end failing us?

Reply:
The default init hasn't bothered be in this case, at least I'm not aware of one. It's this here: https://github.com/gemrb/gemrb/blob/f22f8625ee7676d4a8c32e6091534cbd782ee557/gemrb/plugins/FXOpcodes/FXOpcodes.cpp#L1351

Reply:
These two opcodes [never pass an alpha](https://gibberlings3.github.io/iesdp/opcodes/bgee.htm#op8), so we can fix it in them. I suggest either doing it manually or by adding a  Color::FromBGR, where we force the alpha. There's several other calls that pass a custom Color that need to be updated.

Reply:
Ok, I'll take the `FromBGR` idea.

Reply:
I guess you meant all 4 methods on 0x8 and 0x9, right?

Reply:
Yeah, those are fine, but aren't the other SetColorMod calls problematic too? I don't see anyone setting a non-zero alpha.

Reply:
This way I doesn't even work for my known case any more... Something eats the opaque value between opcode and `applyMod`.

Reply:
It's not some other call resetting it to its value?

Reply:
At a second thought, I think the initial idea is better and the 2nd approach is fruitless: This `applyMod` isn't designed to work on any alpha value. Instead, it takes the original RGB, applies some mod and sets the destination's RGB. And yes, by any default init. of `Color`, alpha is 0, so is the palette's setup buffer.

So in my view, `applyMod` is the place to mark these colors as opaque.

Reply:
It feels like an unnecessary side effect, since `dest` is passed by reference. `GlobalColorMod` does not use alpha, so you could fix its `rgb` member instead, it's just 4 places.

Reply:
That doesn't fix this since it is not copied on all code paths, on some it's just source.rgb x mod.rgb -> dest.rgb. So either we set dest.a = 0xFF (as initially) inside of `applyMod`, or we set buffer.a = dest.a = 0xFF outside of `applyMod`, or we change the color init., but I'm not sure about the impact by this.

Reply:
Or buffer.a = dest.a = src.a

Reply:
That's good we rule out unexpected side effects but also don't have to rewrite things or do another set-alpha-run.

Reply:
What other case does that (I admit I forgot the one ColorMods.rgb setter in the same function)?

Reply:
Look at everything in `applyMod` that does not end in `dest = src;`: you will only see color modifications. So if `dest.a` is blank, it will remain regardless of source.a and modifier.a.